File: spaghetti-typescript-support/src/main/resources/Spaghetti.ts
interface Spaghetti {
	[s: string]: any;

	/**
	 * Returns the module version.
	 */
	getModuleVersion():string;

	/**
	 * Returns the Spaghetti version used to build the module.
	 */
	getSpaghettiVersion():string;

	/**
	 * Returns the name of the module.
	 */
	getModuleName():string;

	/**
	 * Returns a URL pointing to this module's given resource.
	 */
	getResourceUrl(resource:string):string;
}

declare var Spaghetti: Spaghetti;

File: spaghetti-kotlin-support/src/main/resources/Spaghetti.kt
public native object Spaghetti {
	/**
	 * Returns the module version.
	 */
	fun getModuleVersion(): String = noImpl

	/**
	 * Returns the Spaghetti version used to build the module.
	 */
	fun getSpaghettiVersion(): String = noImpl

	/**
	 * Returns the name of the module.
	 */
	fun getModuleName(): String = noImpl

	/**
	 * Returns a URL pointing to this module's given resource.
	 */
	fun getResourceUrl(resource:String): String = noImpl
}
